### ISSUE TITLE
Can you compress the xnb files even more?

### ISSUE THREAD

**Issue by Spaceman333** [LeonBlade/xnbcli#4]
I'm making a mod for Cryptark, a 2D space mech game made by Alientrap.

I used this xnbcli tool to unpack the .png files + a .json file that were inside. I edited the grapics by simply changing the color. Then I repacked it with the tool. It all works wonderfully and even shows up in the game successfully but I noticed something wierd.

Original .xnb file was 13kb
New repacked .xnb file is 62kb

Its almost six times as big.

I was wondering if there is a way to compress the filesize even more?

**Comment by LeonBlade** [comment]
Glad to see you found some use in this tool.  The reason the file is so much bigger is because while my tool has implementation for decompression of the LZX format, it does not currently support compression.  Unfortunately, without having the codebase for LZX compression for reference, it's difficult to recreate the compression algorithm from scratch.

I was going to consider using the official binaries for compression if you were on Windows, but I haven't gotten around to it yet.

**Comment by Spaceman333** [comment]
Yeah actually thank you for making this tool, I was really happy to be able to access and edit the contents of the xnb files that I didn't know how to work with. It significantly enabled a whole lot more options of what I can do in it that weren't an option before. Not to mention how easy it is to use. Really appreciate that.

I quickly skimmed over the description of the tool, saw LZX compression, so I assumed that this tool has compression, but perhaps a different format than what was used for the original xnb files I worked with. I didn't even realize there was no compression lol.

Thats actually relieving to hear, since my main concern was that the different filesize of the new repacked xnb file made me think "uh oh is this the right format? will this introduce performance issues in the game?". But since its just uncompressed, I think that just simply means the only difference is that the png files inside the xnb file will load quicker during startup at the cost of larger filesize on the hd. Thats totally fine by me.

Thank you.
